### ISSUE TITLE
CT: coin term not in expression

### ISSUE THREAD

**Issue by KatieWoe** [phetsims/expression-exchange#139]
```
expression-exchange : xss-fuzz : run
Uncaught Error: Assertion failed: coin term being released is not in expression being edited, this should not occur
Error: Assertion failed: coin term being released is not in expression being edited, this should not occur
    at window.assertions.assertFunction (https://bayes.colorado.edu/continuous-testing/snapshot-1541492814290/assert/js/assert.js:22:13)
    at Array.coinTermUserControlledListener (https://bayes.colorado.edu/continuous-testing/snapshot-1541492814290/expression-exchange/js/common/model/ExpressionManipulationModel.js?bust=1541495522448:759:23)
    at Emitter.emit3 (https://bayes.colorado.edu/continuous-testing/snapshot-1541492814290/axon/js/Emitter.js?bust=1541495522448:259:52)
    at Property._notifyListeners (https://bayes.colorado.edu/continuous-testing/snapshot-1541492814290/axon/js/Property.js?bust=1541495522448:277:29)
    at Property.setValueAndNotifyListeners (https://bayes.colorado.edu/continuous-testing/snapshot-1541492814290/axon/js/Property.js?bust=1541495522448:258:14)
    at Property.set (https://bayes.colorado.edu/continuous-testing/snapshot-1541492814290/axon/js/Property.js?bust=1541495522448:201:16)
    at Object.endDrag (https://bayes.colorado.edu/continuous-testing/snapshot-1541492814290/expression-exchange/js/common/view/AbstractCoinTermNode.js?bust=1541495522448:382:48)
    at MovableDragHandler.movableDragHandlerEnd (https://bayes.colorado.edu/continuous-testing/snapshot-1541492814290/scenery-phet/js/input/MovableDragHandler.js?bust=1541495522448:89:15)
    at MovableDragHandler.endDrag (https://bayes.colorado.edu/continuous-testing/snapshot-1541492814290/scenery/js/input/SimpleDragHandler.js?bust=1541495522448:269:26)
    at Object.up (https://bayes.colorado.edu/continuous-testing/snapshot-1541492814290/scenery/js/input/SimpleDragHandler.js?bust=1541495522448:124:16)
id: Bayes Chrome
Approximately 11/6/2018, 1:26:54 AM
```

**Comment by pixelzoom** [comment]
As of 3/28/19, this is still happening sporadically.

**Comment by jbphet** [comment]
I've tried to duplicate this by fuzz testing locally, and I never hit it.  I also tried a number of use cases with the timing of expression editing and, again, I can't make it happen.  However, it's fairly easy to make it happen as a multi-touch issue by using the following sequence: 
1. create an expression (any of the first three screens can be used)
2. drag a separate coin term out of the creator box and hold it away from the created expression
3. put the expression into edit mode
4. release the separate coin term

The assert is there to make sure that if a coin term is released by the user while an expression is being edited, that particular coin term is a part of the expression.  That's because once an expression is being edited, access to all other coin terms becomes disabled, so the condition should never occur.

I suspect this problem is due to some very quick and rare race condition.  I've decided to just handle the situation where a coin term is released that isn't a part of the expression being edited, since it should prevent the CT failure and also resolves the multi-touch issue.
